### ISSUE TITLE
Add support for Gnome 43

### ISSUE THREAD

**Issue by steve-todorov** [F-i-f/soft-brightness#58]
Are there any plans for Gnoem 43 support ? :)

**Comment by fastrizwaan** [comment]
Yes, please update for gnome 43, Please!

**Comment by fastrizwaan** [comment]
Friends, till Gnome 43 is supported, we can use https://github.com/LordAmit/Brightness

`pip install brightness-controller-linux`

`brightness-controller`

**Comment by Turbine1991** [comment]
> Friends, till Gnome 43 is supported, we can use https://github.com/LordAmit/Brightness
> 
> `pip install brightness-controller-linux`
> 
> `brightness-controller`

Thank you.

On Fedora I had to do this beforehand in order for it to build correctly:
`sudo dnf install python3-devel ddcutil`

Unfortunately it didn't work on desktop. The toggles do nothing.

**Comment by fastrizwaan** [comment]
friends another extension exists, https://extensions.gnome.org/extension/4167/custom-hot-corners-extended/

We can set shortcuts to that...
![Screenshot from 2022-10-28 21-08-38](https://user-images.githubusercontent.com/1161682/198831160-49442598-0d6b-4a50-aeeb-d6b601226f98.png)

**Comment by Turbine1991** [comment]
> friends another extension exists, https://extensions.gnome.org/extension/4167/custom-hot-corners-extended/
> 
> We can set shortcuts to that... ![Screenshot from 2022-10-28 21-08-38](https://user-images.githubusercontent.com/1161682/198831160-49442598-0d6b-4a50-aeeb-d6b601226f98.png)

Good one, it works a treat on my desktop. Thanks! This extension seems to be a whole lot more than just hot corners.

**Comment by fastrizwaan** [comment]
![image](https://user-images.githubusercontent.com/1161682/199489716-556c59db-adb9-4ef4-b792-8b2f02c01d14.png)

Gnome Accessiblity - > zoom ->Color effects can be used directly

**Comment by sergiotca** [comment]
I'm pretty used to adjusting my brightness setting throughout the day with this extension. It is just so easy to use. I cannot wait for it to get ported to 43.

I appreciate both options [fastrizwaan](https://github.com/fastrizwaan) suggests, but I'd rather have the brightness slider right up there on the Gnome Shell menu instead of having to drill down the settings menu several times a day, or learning a new key combo to make use of the [Hot Corners](https://extensions.gnome.org/extension/4167/custom-hot-corners-extended/) extension extended functionality (which I guess I'll have to do in the mean time since it seems to be the most sensible solution for now).

**Comment by fastrizwaan** [comment]
Yes, I also use soft brightness slider always. I am really missing it. Waiting for it. I am daily visiting this github repo to check on updates.

**Comment by fastrizwaan** [comment]
Can someone make a slider for gnome-shell? One can get the values from gsettings and change them as required.
```
rizvan@debian:~$ cat set_brighness.sh 

gsettings set org.gnome.desktop.a11y.magnifier mag-factor 1 ;# do not magnify
gsettings set org.gnome.desktop.a11y.magnifier brightness-red $1
gsettings set org.gnome.desktop.a11y.magnifier brightness-green $1
gsettings set org.gnome.desktop.a11y.magnifier brightness-blue $1
gsettings set org.gnome.desktop.a11y.applications screen-magnifier-enabled true ;#enable accessiblity

```
Usage:
```
rizvan@debian:~$ sh set_brighness.sh -0.80 ; # 20% brightness
rizvan@debian:~$ sh set_brighness.sh -0.70 ; # 30% brightness
rizvan@debian:~$ sh set_brighness.sh -0.50 ; # 50% brightness
rizvan@debian:~$ sh set_brighness.sh -0.00 ; # 100% brightness

```

![Screenshot from 2022-11-15 23-36-35](https://user-images.githubusercontent.com/1161682/201995148-eaaa240f-1886-41b1-9f48-c14a82fc78da.png)

**Comment by fastrizwaan** [comment]
Please try: https://github.com/fastrizwaan/gnome-set-brightness

**Comment by azzamsa** [comment]
https://github.com/daitj/gnome-display-brightness-ddcutil works well for me.

**Comment by zh54tg234f** [comment]
im desperate all other solution either don't work for me or look like crap.

i tried fixing but i got :

![nuh](https://user-images.githubusercontent.com/117942238/203073094-60c63694-1879-4e6c-b0a0-e675461b4d28.png)

i have sadly no experience nor clue how to debug it. (sadly limited time to look into it)

white knight, where are you ? i need to proper dim my screen to fight climate change lmao

**Comment by jkitching** [comment]
I have created a pull request with GNOME 43 support here:
https://github.com/F-i-f/soft-brightness/pull/59

In the meantime, feel free to clone the forked repository and follow the build instructions:
https://github.com/jkitching/soft-brightness

**Comment by va1entin** [comment]
FYI

While there was no fix for Gnome 43 I stumbled upon a tool called [`ddccontrol`](https://github.com/ddccontrol/ddccontrol) that changes your actual screen brightness (without a fake dim effect). It offers a GUI and CLI.

While it of course doesn't integrate as nicely into Gnome as soft-brightness it might still be interesting to some (and works independently of your desktop environment).

**Comment by fastrizwaan** [comment]
> FYI
> 
> While there was no fix for Gnome 43 I stumbled upon a tool called [`ddccontrol`](https://github.com/ddccontrol/ddccontrol) that changes your actual screen brightness (without a fake dim effect). It offers a GUI and CLI.
> 
> While it of course doesn't integrate as nicely into Gnome as soft-brightness it might still be interesting to some (and works independently of your desktop environment).

Thankfully gddccontrol works with regular monitor (VESA) too. will be useful for gnome 44 ;-) till then @jkitching is the man!

**Comment by newinnov** [comment]
> I have created a pull request with GNOME 43 support here: #59
> 
> In the meantime, feel free to clone the forked repository and follow the build instructions: https://github.com/jkitching/soft-brightness

Would it be possible for you to share releases too? As owner has either abandoned this or took better things in Life ..

**Comment by jkitching** [comment]
Hi @F-i-f, I saw you made some commits a couple days ago.  Any chance you could take a look at pull request #59 for GNOME 43 support?

**Comment by jkitching** [comment]
@newinnov here's a build of Soft Brightness which includes GNOME 43 support in the meantime.
[soft-brightness_gnome-43-support.zip](https://github.com/F-i-f/soft-brightness/files/10330435/soft-brightness_gnome-43-support.zip)

**Comment by newinnov** [comment]
> @newinnov here's a build of Soft Brightness which includes GNOME 43 support in the meantime. [soft-brightness_gnome-43-support.zip](https://github.com/F-i-f/soft-brightness/files/10330435/soft-brightness_gnome-43-support.zip)

Thank you @jkitching for sharing build.  It will be very useful for anyone who comes across this thread. I reverted back to Ubuntu LTS from bi-annual release, hence not able to test it right now. Regardless, it will be very useful for other on same boat.

**Comment by milanseo** [comment]
Is someone going to update this extension for gnome 43 already? I'm using 3 monitors and it is a chore to have them on good brightness level, and changing that in the evening and in the mornings. This extension was a God send.

**Comment by sheepdestroyer** [comment]
> Is someone going to update this extension for gnome 43 already? I'm using 3 monitors and it is a chore to have them on good brightness level, and changing that in the evening and in the mornings. This extension was a God send.

jkitching did ; it works great. Look at two posts over yours.

**Comment by sheepdestroyer** [comment]
@jkitching 
Could you maybe publish your fork on https://extensions.gnome.org/ ?

**Comment by fastrizwaan** [comment]
now gnome 44 support?

**Comment by jkitching** [comment]
Hey @F-i-f, since there is still demand for a working version of your excellent extension, I am going to start making releases on the forked repo, and submit an entry to the official GNOME extensions site.  Please let me know if you want to move development back to this repo!

---

I've updated the fork at https://github.com/jkitching/soft-brightness to support GNOME 44.

Here is the "official" v31 release:
https://github.com/jkitching/soft-brightness/releases/tag/v31

I've also uploaded the extension to the official GNOME extensions directory website:
https://extensions.gnome.org/extension/5880/soft-brightness/
The URL won't work quite yet, since it is still under review.

**Comment by azzamsa** [comment]
@jkitching IMHO, the issue tab needs to be enabled in your forked repo.

**Comment by jkitching** [comment]
I didn't realize the issues tab had to be explicitly enabled (: I've added
it, thanks!
